user-guide/Advanced_Modules/User Defined APIs/UD_APIs_Troubleshooting.md

```diff
@@ -0,0 +1,118 @@
+
+## Issues when triggering user-defined APIs
+
+If issues occur when you trigger a user-defined API, follow the steps below to resolve them.
+
+1. Make sure the **IIS rewrite rule** is set correctly.
+
+   Verify if the IIS rewrite rule that should redirect incoming HTTP requests to the UserDefinableEndpoint DxM contains the correct configuration. Make sure that the port in this rule matches the port used by the DxM. For more information, see [Kestrel](xref:UD_APIs_UserDefinableApiEndpoint#kestrel).
+
+1. Make sure **bindings are created in IIS**.
+
+   To be able to trigger a user-defined API, you need to make sure one or more bindings are defined in IIS alongside the rewrite rule:
+
+   1. Open the *Internet Information Services (IIS) Manager* app in Windows.
+
+   1. In the pane on the left, select *Sites* > *Default Web Site*.
+
+   1. In the pane on the right, under *Edit Site*, click *Bindings*
+
+   1. In the pop-up window, check the bindings. If necessary, create a binding for HTTP and/or HTTPS, and select the certificate, IP address, or hostname. For more information, see [Setting up HTTPs on a DMA](xref:Setting_up_HTTPS_on_a_DMA).
+
+1. Make sure **proxy is enabled in IIS**.
+
+   1. Open the *Internet Information Services (IIS) Manager* app in Windows.
+
+   1. In the pane on the left, select the server. This should be the top item in the tree.
+
+   1. In the center pane, double-click *Application Request Routing Cache*.
+
+   1. In the pane on the right, under *Proxy*, click *Server Proxy Settings*.
+
+   1. Make sure that the top checkbox *Enable proxy* is selected.
+
+1. Make sure the **UserDefinableApiEndpoint DxM is running**.
+
+   The UserDefinableApiEndpoint DxM should always be running. To check this, open Windows Task Manager and check whether the process called *DataMiner UserDefinableApiEndpoint* is running. You can also check the *Services* tab to see if the service with the same name has the *Running* status.
+
+   If the DxM is not running, try to start the service by right-clicking it and selecting *Start*. If the service immediately stops again, this could be because the configured port is used by another process, or because the installation is corrupt. Check the logging to find out more. If the installation is corrupt, try to install the DxM again manually.
+
+1. Make sure the **UserDefinableApiEndpoint DxM and the manager in SLNet are connected to NATS**.
+
+   To allow communication between the DxM and the manager in SLNet, a connection to the NATS message bus is required.
+
+   - For the DxM, open the [UserDefinableApiEndpointInstaller.txt](#logging) log file and check whether the log line with message "Successfully initialized the MessageBrokerWrapper" is displayed after the most recent startup. There may be errors before this that indicate that the creation of the connection had to be retried. This is fine as long as the success message is logged at the end.
```

Comment:
The file should be "UserDefinableApiEndpoint.txt" instead of the one mentioned now